logging/google/cloud/logging/handlers/app_engine.py

```diff
@@ -27,47 +27,63 @@
 # /appengine-vmruntime/vmruntime/cloud_logging.py
 
 import logging.handlers
-import os
 
-from google.cloud.logging.handlers._helpers import format_stackdriver_json
+from google.cloud.logging.handlers.handlers import CloudLoggingHandler
+from google.cloud.logging.handlers.transports import BackgroundThreadTransport
+from google.cloud.logging.resource import Resource
 
-_LOG_PATH_TEMPLATE = '/var/log/app_engine/app.{pid}.json'
-_MAX_LOG_BYTES = 128 * 1024 * 1024
-_LOG_FILE_COUNT = 3
+DEFAULT_LOGGER_NAME = 'python'
```

Comment:
The logName will be automatically set to 'projects/PROJECT_ID/logs/{}'.format(LOG_NAME), and the LOG_NAME is user specified in 'logger = client.logger(LOG_NAME)'. I saw your example logName is end up with '.shutdown', and there should be other kinds of the log name suffix. Are there any way to detect the logName suffix and set it?